docs/topics/db/aggregation.txt

```diff
@@ -340,29 +340,63 @@ Order of ``annotate()`` and ``filter()`` clauses
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 When developing a complex query that involves both ``annotate()`` and
-``filter()`` clauses, particular attention should be paid to the order
-in which the clauses are applied to the ``QuerySet``.
-
-When an ``annotate()`` clause is applied to a query, the annotation is
-computed over the state of the query up to the point where the annotation
-is requested. The practical implication of this is that ``filter()`` and
-``annotate()`` are not commutative operations -- that is, there is a
-difference between the query::
-
-    >>> Publisher.objects.annotate(num_books=Count('book')).filter(book__rating__gt=3.0)
-
-and the query::
-
-    >>> Publisher.objects.filter(book__rating__gt=3.0).annotate(num_books=Count('book'))
-
-Both queries will return a list of publishers that have at least one good
-book (i.e., a book with a rating exceeding 3.0). However, the annotation in
-the first query will provide the total number of all books published by the
-publisher; the second query will only include good books in the annotated
-count. In the first query, the annotation precedes the filter, so the
-filter has no effect on the annotation. In the second query, the filter
-precedes the annotation, and as a result, the filter constrains the objects
-considered when calculating the annotation.
+``filter()`` clauses, pay particular attention to the order in which the
+clauses are applied to the ``QuerySet``.
+
+When an ``annotate()`` clause is applied to a query, the annotation is computed
+over the state of the query up to the point where the annotation is requested.
+The practical implication of this is that ``filter()`` and ``annotate()`` are
+not commutative operations.
+
+Given:
+
+* Publisher A has two books with ratings 4 and 5
+* Publisher B has two books with ratings 1 and 4
+* Publisher C has one book with rating 1
+
+Here's an example with the ``Count`` aggregate::
+
+    >>> a, b = Publisher.objects.annotate(num_books=Count('book')).filter(book__rating__gt=3.0)
+    >>> a, a.num_books
+    (<Publisher: A>, 4)
+    >>> b, b.num_books
+    (<Publisher: B>, 2)
+
+    >>> a, b = Publisher.objects.filter(book__rating__gt=3.0).annotate(num_books=Count('book'))
+    >>> a, a.num_books
+    (<Publisher: A>, 2)
+    >>> b, b.num_books
+    (<Publisher: B>, 1)
+
+Both queries return a list of publishers that have at least one book with a
+rating exceeding 3.0, hence publisher C is excluded. However, the annotation in
+the first query gives an incorrect result for the number of books for each
+publisher because it counts over the number of rows in the joined table
+(publishers left outer joined with books). The second query counts the number
```

Comment:
I think this sentence ("However, ...") needs some clarification. I think we should be leaning towards language that highlights that annotate-filter only filters the already-annotated result, where the opposite is true for filter-annotate. I don't think it's necessary to call out left joins here at all, and would probably just confuse further. Saying that the result is incorrect isn't great because it sounds like Django is doing the wrong thing, even though you clarify with that the intent of the query was meaning to be.

I'm not sure how I'd write this part, but they're my thoughts.